jazkarta/shop/cart.py

```diff
@@ -240,7 +240,10 @@ def save(self):
         """
         self.data['items'] = self._items
         if self.storage_id:
-            storage.set_shop_data([self.storage_id, 'cart'], self.data)
+            if len(self.data['items']):
+                storage.set_shop_data([self.storage_id, 'cart'], self.data)
+            else:
+                storage.del_shop_data([self.storage_id, 'cart'])
 
     def clear(self):
```

Comment:
I do wonder if this should call `.save()`, rather than relying on the caller of this to also call `.save()`.